atecc608a_se.c

```diff
@@ -0,0 +1,226 @@
+        }                                                   \
+    } while(0)
+
+#define ATCAB_DEINIT()    \
+    do                    \
+    {                     \
+        atcab_release();  \
+    } while(0)
+
+#define ASSERT_STATUS(expression, expected, psa_error)          \
+    do                                                          \
+    {                                                           \
+        ATCA_STATUS ASSERT_result = (expression);               \
+        ATCA_STATUS ASSERT_expected = (expected);               \
+        if ((ASSERT_result) != (ASSERT_expected))               \
+        {                                                       \
+            status = (psa_error);                               \
+            goto exit;                                          \
+        }                                                       \
+        status = PSA_SUCCESS;                                   \
+    } while(0)
+
+#define ASSERT_SUCCESS(expression) ASSERT_STATUS(expression,ATCA_SUCCESS, \
+                                      atecc608a_to_psa_error(ASSERT_result))
+
+ATCAIfaceCfg atca_iface_config = {
+    .iface_type = ATCA_I2C_IFACE,
+    .devtype = ATECC608A,
+    .atcai2c.slave_address = 0xC0,
+    .atcai2c.bus = 2,
+    .atcai2c.baud = 400000,
+    .wake_delay = 1500,
+    .rx_retries = 20,
+};
+
+static psa_status_t atecc608a_to_psa_error(ATCA_STATUS ret)
+{
+    switch (ret)
+    {
+    case ATCA_SUCCESS:
+    case ATCA_RX_NO_RESPONSE:
+    case ATCA_WAKE_SUCCESS:
+        return PSA_SUCCESS;
+    case ATCA_BAD_PARAM:
+    case ATCA_INVALID_ID:
+        return PSA_ERROR_INVALID_ARGUMENT;
+    case ATCA_ASSERT_FAILURE:
+        return PSA_ERROR_TAMPERING_DETECTED;
+    case ATCA_SMALL_BUFFER:
+        return PSA_ERROR_BUFFER_TOO_SMALL;
+    case ATCA_RX_CRC_ERROR:
+    case ATCA_RX_FAIL:
+    case ATCA_STATUS_CRC:
+    case ATCA_RESYNC_WITH_WAKEUP:
+    case ATCA_PARITY_ERROR:
+    case ATCA_TX_TIMEOUT:
+    case ATCA_RX_TIMEOUT:
+    case ATCA_TOO_MANY_COMM_RETRIES:
+    case ATCA_COMM_FAIL:
+    case ATCA_TIMEOUT:
+    case ATCA_TX_FAIL:
+    case ATCA_NO_DEVICES:
+        return PSA_ERROR_COMMUNICATION_FAILURE;
+    case ATCA_UNIMPLEMENTED:
+        return PSA_ERROR_NOT_SUPPORTED;
+    case ATCA_ALLOC_FAILURE:
+        return PSA_ERROR_INSUFFICIENT_MEMORY;
+    case ATCA_BAD_OPCODE:
+    case ATCA_CONFIG_ZONE_LOCKED:
+    case ATCA_DATA_ZONE_LOCKED:
+    case ATCA_NOT_LOCKED:
+    case ATCA_WAKE_FAILED:
+    case ATCA_STATUS_UNKNOWN:
+    case ATCA_STATUS_ECC:
+    case ATCA_STATUS_SELFTEST_ERROR:
+    case ATCA_CHECKMAC_VERIFY_FAILED:
+    case ATCA_PARSE_ERROR:
+    case ATCA_FUNC_FAIL:
+    case ATCA_GEN_FAIL:
+    case ATCA_EXECUTION_ERROR:
+    case ATCA_HEALTH_TEST_ERROR:
+    case ATCA_INVALID_SIZE:
+    default:
+        return PSA_ERROR_HARDWARE_FAILURE;
+    }
+}
+
+psa_status_t atecc608a_get_serial_number(uint8_t* buffer, size_t buffer_size,
```

Comment:
Why add a serial number getting function as a public function? Seems unrelated to implementing signature and key exporting, at least. This is potentially useful for verifying that the hardware is reachable and returns something when asked, but I don't see us needing this right now; it's just a fun debug print in an example.